lib/rspec/support/object_inspector.rb

```diff
@@ -13,11 +13,18 @@ def self.inspect(object)
           "#{object.to_s 'F'} (#{object.inspect})"
         elsif RSpec::Support.is_a_matcher?(object) && object.respond_to?(:description)
           object.description
+        elsif object.class == RSpec::Mocks::Double
+          "Double (#{object.instance_variable_get(:@name)})"
         else
           object.inspect
         end
       end
 
+      def self.register(double)
+        raise "Please stub 'name' message on your double" unless double.respond_to?(:name)
+        "Double (#{double.name})"
+      end
```

Comment:
This assumes that the argument will be an rspec-mocks test double.  The point of `register` is that it provides a way for other libraries like rspec-mocks to register how their objects will be presented by `ObjectInspector` w/o `ObjectInspector` having any knowledge of them.  So it should be something like:

``` ruby
def self.register(klass, &block)
  registered_klasses[klass] = block
end

def self.registered_klasses
  @registered_klasses ||= {}
end
```

Then in the `inspect` method above, in the `else` case (that currently has `object.inspect`), it should be something like:

``` ruby
registered_klasses.each do |klass, inspector|
  return inspector.call(object) if klass === object
end

object.inspect
```

In that way, `ObjectInspector` will have zero knowledge of rspec-mocks but exposes an interface that rspec-mocks can use to register how test doubles should be presented.  Does that make sense?